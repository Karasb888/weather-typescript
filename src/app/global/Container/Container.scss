.container {
    position: absolute;
    box-sizing: border-box;
    width: 100vw;
    height: 100vh;
    padding-top: 25px;
    left: 0;
    top: 0;
    overflow: hidden;
    background-image: url('../../../img/background_home.png');
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
}