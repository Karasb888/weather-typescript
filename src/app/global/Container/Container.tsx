import * as React from 'react';
import * as styles from './Container.scss';

const Container: React.FC = ({ children }) => {
    return (
        <div className={styles.container}>
            { children }
        </div>
    );
};

export default Container;
