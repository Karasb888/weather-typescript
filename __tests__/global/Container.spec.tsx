import * as React from 'react';
import { shallow } from 'enzyme';
import Container from '../../src/app/global/Container/Container';

const MockedChild: React.FC = () => <div></div>;

describe('++++ <Container /> ++++', () => {
    it('Renders correctly without child component', () => {
        const wrapper = shallow(<Container />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Renders correctly with child component', () => {
        const wrapper = shallow(<Container><MockedChild /></Container>);
        expect(wrapper).toMatchSnapshot();
    });
});
