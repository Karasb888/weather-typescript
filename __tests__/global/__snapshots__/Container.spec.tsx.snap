// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`++++ <Container /> ++++ Renders correctly with child component 1`] = `
<div
  className="container"
>
  <MockedChild />
</div>
`;

exports[`++++ <Container /> ++++ Renders correctly without child component 1`] = `
<div
  className="container"
/>
`;
